pvlib/snowcoverage.py

```diff
@@ -0,0 +1,229 @@
+        designed for ground snowdepth only. (cm/hr or cm)
+
+    snow_data_type : string
+        Defines what type of data is being passed as snow_data. Acceptable
+        values are "snowfall" and "snow_depth". "snowfall" will be in units of
+        cm/hr. "snow_depth" is in units of cm.
+
+    poa_irradiance : numeric
+        Total in-plane irradiance (W/m^2)
+
+    temperature : numeric
+        Ambient air temperature at the surface (C)
+
+    surface_tilt : numeric
+        Surface tilt angles in decimal degrees. Tilt must be >=0 and
+        <=180. The tilt angle is defined as degrees from horizontal
+        (e.g. surface facing up = 0, surface facing horizon = 90).
+
+    time_step_hours: float
+        Period of the data in hours. (hours between data points)
+
+    sliding coefficient : numeric
+        Another empirically determined coefficient used in [1]. It determines
+        how much snow slides off of the panel if a sliding event occurs.
+    m : numeric
+        A coefficient used in the model described in [1]. It is an
+        empirically determined value given in W/m^2.
+
+    Returns
+    -------
+    snow_coverage : numeric
+        The fraction of a module covered by snow at each time step.
+    '''
+
+    full_coverage_events = fully_covered_panel(snow_data,
+                                               time_step_hours=time_step_hours,
+                                               snow_data_type=snow_data_type)
+    snow_coverage = pd.Series(np.full(len(snow_data), np.nan))
+    snow_coverage = snow_coverage.reindex(snow_data.index)
+    snow_coverage[full_coverage_events] = 1
+    slide_events = snow_slide_event(poa_irradiance, temperature)
+    slide_amount = snow_slide_amount(surface_tilt, sliding_coefficient)
+    slidable_snow = ~np.isnan(snow_coverage)
+    while(np.any(slidable_snow)):
+        new_slides = np.logical_and(slide_events, slidable_snow)
+        snow_coverage[new_slides] -= slide_amount*.1
+        new_snow_coverage = snow_coverage.fillna(method="ffill", limit=1)
+        slidable_snow = np.logical_and(~np.isnan(new_snow_coverage),
+                                       np.isnan(snow_coverage))
+        slidable_snow = np.logical_and(slidable_snow, new_snow_coverage > 0)
+        snow_coverage = new_snow_coverage
+
+    new_slides = np.logical_and(slide_events, slidable_snow)
+    snow_coverage[new_slides] -= slide_amount
+
+    snow_coverage = snow_coverage.fillna(method="ffill")
+    snow_coverage = snow_coverage.fillna(value=0)
+    snow_coverage[snow_coverage < 0] = 0
+    return snow_coverage
+
+
+def DC_loss_factor(snow_coverage, num_strings_per_row):
+    '''
+    Calculates the DC loss due to snow coverage. Assumes that if a string is
+    even partially covered by snow, it produces 0W.
+
+    Parameters
+    ----------
+    snow_coverage : numeric
+        The fraction of a module covered by snow at each time step.
+
+    num_strings_per_row: int
```

Comment:
I'd call this num_strings and define it as the number of parallel-connected strings along a row slant height.